backend/src/main/java/gov/cdc/usds/simplereport/api/export/DataHubUploadController.java

```diff
@@ -40,15 +45,39 @@ public String uploadTestEventCSVToDataHub(@RequestParam String apikey,
 
     @GetMapping(value = "/testEvent", produces = {"text/csv"})
     public ResponseEntity<?> exportTestEventCSV(HttpServletResponse response,
-                                                @RequestParam(defaultValue = "") String startupdateby) throws IOException {
+                                                @RequestParam(defaultValue = "") String startupdateby,
+                                                @CookieValue(value = "csvsavedstartupby", defaultValue = "")
+                                                        String startupdatebyCookie) throws IOException {
+        if (startupdateby.length() == 0 && startupdatebyCookie.length() > 0) {
+            // if the value doesn't come from the url and it's available in the cookie, use that.
+            startupdateby = startupdatebyCookie;
+        } else if (startupdateby.compareToIgnoreCase("all") == 0) {
+            startupdateby = DataHubUploaderService.EARLIEST_TIMESTAMP;
+        }
+
         response.setContentType("text/csv");
         DateFormat dateFormatter = new SimpleDateFormat("yyyy-MM-dd_HH-mm-ss");
         String currentDateTime = dateFormatter.format(new Date());
+        // we want to set the next timestamp in the cookie. But cookie headers must be set before
+        // we start writing back out the html body. So preload the csv so we can call getNextTimestamp()
+        String csvString = this._hubuploadservice.creatTestCVSForDataHub(startupdateby);
+
+        String nexttimestamp = this._hubuploadservice.getNextTimestamp();
 
         String headerKey = "Content-Disposition";
         String headerValue = "attachment; filename=testEvents_" + currentDateTime + ".csv";
         response.setHeader(headerKey, headerValue);
-        response.getWriter().print(_hubuploadservice.creatTestCVSForDataHub(startupdateby));
+        if (nexttimestamp.length() > 0) {  // if there were no matches then this will be empty.
+            Cookie cookie = new Cookie("csvsavedstartupby", nexttimestamp);
+            final int SECONDS_TO_EXPIRE_COOKIE = 60 * 60 * 24 * 90;  // 90 days to refresh
+            // if we setSecure(true) for localhost, then setting cookie fails.
+            cookie.setHttpOnly(this._insecurecookies);
```

Comment:
Do we need this to be a JS-accessible cookie in test environments?

Reply:
Excellent point. Because of the declarative nature of Spring, this cookie-name-constant appears in the method parameters as well. One fix could be to make the cookie data value be json and put `{'dev': '[timestamp]', 'prod': '[timestamp]'}` I'm hoping this cookie approach will be disappearing as soon as we get a CRON job doing this tasks, so it's probably not work the effort. But good catch!

But I'll make sure to warn @timothybest-usds